lib/galaxy/tools/verify/interactor.py

```diff
@@ -733,15 +733,13 @@ def verify_tool(tool_id,
                 quiet=False,
                 test_history=None,
                 force_path_paste=False,
-                maxseconds=None,
+                maxseconds=DEFAULT_TOOL_TEST_WAIT,
```

Comment:
I don't agree with this, this will overwrite `testdef.maxseconds`

Reply:
Not with the removal of lines 736-737 in the same commit ;)

Reply:
Hmm, but then we also need to pass maxseconds in `_verify_outputs`, otherwise it'll use the default `testdef.maxseconds`

Reply:
I've rebased the commit with this:
```diff
         resource_parameters = {}
     tool_test_dicts = tool_test_dicts or galaxy_interactor.get_tool_tests(tool_id, tool_version=tool_version)
     tool_test_dict = tool_test_dicts[test_index]
+    tool_test_dict.setdefault('maxseconds', maxseconds)
     testdef = ToolTestDescription(tool_test_dict)
-    if maxseconds is not None:
-        testdef.maxseconds = int(maxseconds)
     _handle_def_errors(testdef)
 
     if test_history is None:
```

i.e. use the `maxseconds` passed to `verify_tool()` only if `tool_test_dict` does not specify already a value for it. Best of both worlds?

Reply:
I'd probably want to override `testdef.maxseconds` in all cases. what do you think of https://github.com/mvdbeek/galaxy/commit/bcb251b4e14a947c567494b5ef5aaba838cf6643 instead ?

Reply:
Then also remove `maxseconds` attribute from `ToolTestDescription` ?
```diff
diff --git a/lib/galaxy/tools/verify/interactor.py b/lib/galaxy/tools/verify/interactor.py
index 757fe2047b..fc5b646a65 100644
--- a/lib/galaxy/tools/verify/interactor.py
+++ b/lib/galaxy/tools/verify/interactor.py
@@ -1038,14 +1038,10 @@ class ToolTestDescription(object):
     def __init__(self, processed_test_dict):
         test_index = processed_test_dict["test_index"]
         name = processed_test_dict.get('name', 'Test-%d' % (test_index + 1))
-        maxseconds = processed_test_dict.get('maxseconds', DEFAULT_TOOL_TEST_WAIT)
-        if maxseconds is not None:
-            maxseconds = int(maxseconds)
 
         self.test_index = test_index
         self.tool_id = processed_test_dict["tool_id"]
         self.name = name
-        self.maxseconds = maxseconds
         self.required_files = processed_test_dict.get("required_files", [])
 
         inputs = processed_test_dict.get("inputs", {})
```

Reply:
For the record, we agreed on Gitter that my latest commit was correct since `maxseconds` can be specified per test in the tool XML (see https://docs.galaxyproject.org/en/latest/dev/schema.html#id77 ).